tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsWriter.cs

```diff
@@ -237,8 +227,10 @@ private async Task ProcessQueueLoopAsync()
                 continue;
             }
 
-            await _processingMutex.WaitAsync().ConfigureAwait(false);
-            _processingMutex.Reset();
+            if (!_completionSource.Task.IsCompleted)
+            {
+                _completionSource.Task.Wait(_waitTimeSpan);
```

Comment:
Using a synchronous `Task.Wait()` in async code is generally not recommended. It can cause deadlocks and by blocking the thread, it can starve the thread pool.

Reply:
Yeah, this is basically a "never ever do this" pattern in our code. It _will_ break someone, practically guaranteed